sw/cheri/common/asm.hh

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdint>
+
+namespace ASM{
+  struct Ibex{
+    static void nop(void) {
+      __asm__ volatile ("nop");
+    }
+
+    static void wfi(void) {
+      __asm__ volatile ("wfi");
+    }
+
+    static uint32_t mtime_read(void ){
+      uint32_t value = 0;
+      __asm__ volatile ("csrr %0, mtime" : "=r"(value));
+      return value;
+    }
+
+    static uint32_t mstatus_read(void ){
+      uint32_t value = 0;
+      __asm__ volatile ("csrr %0, mstatus" : "=r"(value));
+      return value;
+    }
```

Comment:
In sw/cheri/boot/boot_loader.cc, there is a generic CSR read macro. Maybe port it here and use it to implement these functions.

Reply:
Done